Create Media Segments Wiki Page (#1179)
jellyfin/jellyfin.org: docs/general/server/media/media-segments.md

> ---

Reviewer: Would it make more sense to add this in the "metadata" category instead of "media"? Feels like "media" is to highlight the different media types (music/books etc) and folder structures

Author: I also don't really know where to put it besides it should be under server

Reviewer 2: i agree

> ---

Reviewer 2: i am somewhat missing the step-by-step instructions how to enable it.

Reviewer: something like this?

## Getting started

There's a few steps to follow if you want to use media segments.

1. First, you'll need one or multiple plugins that can provide media segments. Read the [plugins](#) section below to find out more.
2. Run the "Media segment scan" task in the dashboard to create segments immediately, this task also runs automatically in the background.
3. Enable actions for the different segment types, the way you do this differs per client but they are generally found in the playback settings of the client.

---
uid: server-media-media-segments
title: Media Segments
---

# Media Segments

> Media segments are sections of media files with a defined type, stored within Jellyfin. They are different from chapters in that chapters don't have any defined type.

Reviewer 3: ```suggestion
Media segments are a type of metadata for media files stored in Jellyfin. Unlike chapters, which have no type, media segements can contain type information, allowing different actions based on the type of a given time segment.
```

## Types

The following types are currently available:

- Commercial
- Preview
- Recap
- Outro
- Intro

## An example

> An video that is 16 minutes (00:16:00) long could have the following segments:

Reviewer 4: ```suggestion
A video that is 16 minutes (00:16:00) long could have the following segments:
```

> 1. from 00:00:00 until 00:03:08 there is an `Intro` segment.

Reviewer 4: ```suggestion
1. From 00:00:00 until 00:03:08 there is an `Intro` segment.
```

> 2. from 00:08:03 until 00:08:59 there is an `Commerical` segment.

Reviewer 4: ```suggestion
2. From 00:08:03 until 00:08:59 there is an `Commercial` segment.
```

> 3. from 00:14:30 until 00:16:00 there is an `Outro` segment.

Reviewer 4: ```suggestion
3. From 00:14:30 until 00:16:00 there is an `Outro` segment.
```

> Jellyfin can store this information and provide it via the MediaSegment Api to clients. Clients can then decide what they wants to do with the provided information, such as displaying a "skip" button to allow convenient skipping.

Reviewer 4: ```suggestion
Jellyfin can store this information and provide it via the MediaSegment API to clients. Clients can then decide what they wants to do with the provided information, such as displaying a "skip" button to allow convenient skipping.
```

## Creating Media Segments

> As of Jellyfin 10.10, media segments can only be created by plugins. There is an official `Chapter Segments Provider` plugin that creates media segments from chapters.

Reviewer 3: I think this is mostly covered below in the "Plugins" section, so perhaps unifying those two bits is better and this can be removed.

Author: moved the plugin mention to the plugins section

## Clients

Media segments are currently supported in the web interface and the Android TV[^1] app.

[^1]: Available starting from 0.18, soon [in beta](/posts/android-betas)

> The server simply delivers the information about where the segment is in the video and of what type it is, it remains up the individual client to decide what to do with that information. Since media segments is a new feature, please do not expect many clients to support it.

Reviewer 3: ```suggestion
The server is only responsible for delivering the segment metadata to the client; it is up to each client to decide what to do with the information. For example, a client may implement a "skip intro" button when it encounters an "intro"-type segment. Because it is still a new feature, client support is very sparse and will take some time to become more wide-spread.
```

## Plugin support

> Plugins can utilize this system to store their information about Intros, Outros, Commercials and all other types of segments. This information can then used by clients to provide actions, such as a "Skip" button in their UI. This approach generalizes how segments are handled, so plugins no longer need to inject javascript into the WebUI to get IntroSkipping, and similar functionality can be achieved without custom modifications to other clients.

Reviewer 4: ```suggestion
Plugins can utilize this system to store their information about Intros, Outros, Commercials and all other types of segments. This information can then be used by clients to provide actions, such as a "Skip" button in their UI. This approach generalizes how segments are handled, so plugins no longer need to inject javascript into the WebUI to get IntroSkipping, and similar functionality can be achieved without custom modifications to other clients.
```

> Plugins can utilize this system to store their information about Intros, Outros, Commercials and all other types of segments. This information can then be used by clients to provide actions, such as a "Skip" button in their UI. This approach generalizes how segments are handled, so plugins no longer need to inject javascript into the WebUI to get IntroSkipping, and similar functionality can be achieved without custom modifications to other clients.

Reviewer 3: ```suggestion
Plugins can now utilize the media segments system to store information about the various types of segments, without having to rely on their own custom implementations or workarounds such as injecting custom Javascript into the WebUI. This system should therefore make it much easier to implement new functionality around segments and ensure that the segment handling is consistent among clients.
```

> This does _NOT_ make the existing plugins obsolete, however as we still rely on them to generate the segments. They would simply have to be updated to work with the media segments system.

Reviewer 3: ```suggestion
Existing plugins should be modified to support the media segments system; while this does not make them obsolete, it does provide more flexibility and Jellyfin will still require such plugins to generate the media segment data in the first place.
```

---
uid: server-media-media-segments

> title: Media Segments

Reviewer: ```suggestion
title: Media segments
```

---

> # Media Segments

Reviewer: ```suggestion
# Media segments
```

Media segments are a type of metadata for media files stored in Jellyfin. Unlike chapters, which have no type, media segements can contain type information, allowing different actions based on the type of a given time segment.

## Types

The following types are currently available:

- Commercial
- Preview
- Recap
- Outro
- Intro

## An example

A video that is 16 minutes (00:16:00) long could have the following segments:

1. From 00:00:00 until 00:03:08 there is an `Intro` segment.
2. From 00:08:03 until 00:08:59 there is a `Commercial` segment.
3. From 00:14:30 until 00:16:00 there is an `Outro` segment.

> Jellyfin can store this information and provide it via the MediaSegment API to clients. Clients can then decide what they want to do with the provided information, such as displaying a "skip" button to allow convenient skipping.

Reviewer: ```suggestion
Jellyfin can store this information and provide it to clients. Clients can then decide what they want to do with the provided information, such as displaying a "skip" button to allow convenient skipping.
```

The API part is too technical

> ## Creating Media Segments

Reviewer: ```suggestion
## Creating media segments
```

> As of Jellyfin 10.10, media segments can only be created by plugins. [Read more](#plugin-support)

Reviewer: ```suggestion
Media segments are provided by plugins. [Read more](#plugin-support)
```

There is no other way and there will be no other way.

Reviewer: At this point I'd also merge it with the plugin support section

## Clients

Media segments are currently supported in the web interface and the Android TV[^1] app.

> [^1]: Available starting from 0.18, soon [in beta](/posts/android-betas)

Reviewer: I'd remove the ATV support bit for now and add it back once released

> The server is only responsible for delivering the segment metadata to the client; it is up to each client to decide what to do with the information. For example, a client may implement a "skip intro" button when it encounters an "intro"-type segment. Because it is still a new feature, client support is very sparse and will take some time to become more wide-spread.

Reviewer: Kinda repeated information from the example section

## Plugin support

> Plugins can utilize this system to store their information about Intros, Outros, Commercials and all other types of segments. This information can then be used by clients to provide actions, such as a "Skip" button in their UI. This approach generalizes how segments are handled, so plugins no longer need to inject Javascript into the web client to get intro skipping, and similar functionality can be achieved without custom modifications to other clients.

Reviewer: ```suggestion
Plugins can utilize this system to store their information about intros, outros, commercials and all other types of segments. This information can then be used by clients to provide actions, such as a "skip" button in their user interface. This approach generalizes how segments are handled, so plugins no longer need to inject JavaScript into the web client to get intro skipping, and similar functionality can be achieved without custom modifications to other clients.
```

Also we shoudn't mention the JavaScript injection part because we don't support such hacks.

Reviewer: > Also we shoudn't mention the JavaScript injection part because we don't support such hacks.

not solved

There is an official `Chapter Segments Provider` plugin that creates media segments from chapters.

> This does _NOT_ make the existing plugins obsolete, as we still rely on them to generate the segments. They would simply have to be updated to work with the media segments system.

Reviewer: Feels like we're defending ourselves here? Why?

Author: Venson wrote this, this was straight from the blog post which Venson asked me to turn into a docs page

Reviewer: I'd remove it, it does not provide any useful information for a user learning about media segments. Instead, you could mention that there are also third-party plugins to provide media segments and link to our documentation page that lists third-party plugin repositories.
